### ISSUE TITLE
Extra correlated variables affect the result

### ISSUE THREAD

**Issue by Jellby** [lebigot/uncertainties#95]
I'd like to create correlated variables (actually physical constants, from https://physics.nist.gov/cuu/Correlations/index.html), but I'm confused by the fact that adding more correlated variables changes the result of operations that *do not include the extra variables* at all.

For instance, consider the example below, where I calculate the difference between `m_n` and `m_p` (neutron and proton mass). If I define only these two variables, I get a reasonable result, but if I define a third correlated variable (although I never use it), the uncertainty in the result grows beyond recognition

```
import uncertainties
import numpy as np

def_m_p = (1.672621898e-27, 0.000000021e-27)
def_m_n = (1.674927471e-27, 0.000000021e-27)
def_w = (5.8789238e10, 0.0000034e10)
def_corr = np.eye(3)
def_corr[0,1] =  0.9992 # m_p, m_n
def_corr[0,2] = -0.0004 # m_p, w
def_corr[1,2] = -0.0004 # m_n, w

def_corr += def_corr.T - np.diag(def_corr.diagonal())

select = [0, 1]
corr = def_corr[np.ix_(select, select)]
(m_p, m_n) = uncertainties.correlated_values_norm([def_m_p, def_m_n], corr)
print('\n{} variables'.format(len(select)))
print(corr)
print(m_n-m_p)

select = [0, 1, 2]
corr = def_corr[np.ix_(select, select)]
(m_p, m_n, w) = uncertainties.correlated_values_norm([def_m_p, def_m_n, def_w], corr)
print('\n{} variables'.format(len(select)))
print(corr)
print(m_n-m_p)
```
This gives me:
```
2 variables
[[1.     0.9992]
 [0.9992 1.    ]]
(2.3055730+/-0.0000008)e-30

3 variables
[[ 1.000e+00  9.992e-01 -4.000e-04]
 [ 9.992e-01  1.000e+00 -4.000e-04]
 [-4.000e-04 -4.000e-04  1.000e+00]]
(0.0+/-1.5)e-11
```

Is this expected? What am I doing wrong?

**Comment by lebigot** [comment]
This result is incorrect, as can be seen by normalizing all the variables to 1±something first, and setting the correct factor back (which amounts to a change of units):

```python
def_m_p_orig = (1.672621898e-27, 0.000000021e-27)
def_m_n_orig = (1.674927471e-27, 0.000000021e-27)
def_w_orig = (5.8789238e10, 0.0000034e10)


# Normalization of the variables:
def_m_p = tuple(value/def_m_p_orig[0] for value in def_m_p_orig)
def_m_n = tuple(value/def_m_n_orig[0] for value in def_m_n_orig)
def_w = tuple(value/def_w_orig[0] for value in def_w_orig)

def_corr = np.eye(3)
def_corr[0,1] =  0.9992 # m_p, m_n
def_corr[0,2] = -0.0004 # m_p, w
def_corr[1,2] = -0.0004 # m_n, w

def_corr += def_corr.T - np.diag(def_corr.diagonal())

select = [0, 1]
corr = def_corr[np.ix_(select, select)]
(m_p, m_n) = uncertainties.correlated_values_norm([def_m_p, def_m_n], corr)

# Normalizing the variables back:
m_p *= def_m_p_orig[0]
m_n *= def_m_n_orig[0]

print('\n{} variables'.format(len(select)))
print(corr)
print(m_n-m_p)

select = [0, 1, 2]
corr = def_corr[np.ix_(select, select)]
(m_p, m_n, w) = uncertainties.correlated_values_norm([def_m_p, def_m_n, def_w], corr)

# Normalizing the variables back:
m_p *= def_m_p_orig[0]
m_n *= def_m_n_orig[0]
w *= def_w_orig[0]

print('\n{} variables'.format(len(select)))
print(corr)
print(m_n-m_p)
```
gives
```

2 variables
[[1.     0.9992]
 [0.9992 1.    ]]
(2.3055730+/-0.0000008)e-30

3 variables
[[ 1.000e+00  9.992e-01 -4.000e-04]
 [ 9.992e-01  1.000e+00 -4.000e-04]
 [-4.000e-04 -4.000e-04  1.000e+00]]
(2.3055730+/-0.0000008)e-30
```

I am wondering whether this doesn't come from the wildly differing orders of magnitude between the masses `m_n` and `m_p`, and `w`.

In any case, the calculation method above (normalization to 1 and then back to the units of choice) suggests a solution that could be implemented in the code.

I'll have a look.

**Comment by lebigot** [comment]
I'm confirming my guess: the calculation is currently done by first building the covariance matrix (https://github.com/lebigot/uncertainties/blob/91397f58a8211933cc52faf311dafc1fd08e90bc/uncertainties/core.py#L221), and this matrix is ill-conditioned because of the wildly different orders of magnitude of the standard deviations of the three variables.

I am sure that I can whip up a solution.

A warning against matrices that can yield incorrect numerical results should also be added to the documentation after this is done.

**Comment by lebigot** [comment]
This problem is [solved in version 3.1+](https://pypi.org/project/uncertainties/).
